nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -360,6 +371,38 @@ Tensor Pooling2DLayer::pooling2d(Tensor &in, bool training, Tensor &output) {
     break;
   }
   case PoolingType::global_max: {
+    pool_fn = [&, this](const float *in_data, int start_h, int start_w) {
+      int end_h = start_h + patch_height;
+      int end_w = start_w + patch_width;
+
+      float max_val = std::numeric_limits<float>::lowest();
+
+      decltype(max_idx_global)::iterator indexor;
+      if (training) {
+        max_idx_global.emplace_back();
+        indexor = max_idx_global.end() - 1;
+      }
+
+      for (int h = start_h; h < end_h; ++h) {
+        for (int w = start_w; w < end_w; ++w) {
+          int cur_idx = h * in_width + w;
+          float val = in_data[cur_idx];
+          if (max_val < val) {
```

Comment:
how about just calculate max_val and its idx, store them in local variables, and then later add to max_idx_global at once. Current checks are heavily error-prone and tough to reason.

Reply:
I agree this is heavily error-prone. Best way I can think of is using lambda but this makes the loop 2x slower (tested when building conv2d). Could you suggest me better way? I'll ask offline